### ISSUE TITLE
setuptools option include_package_data requires preserve_egg_dir since conda build 1.21.9

### ISSUE THREAD

**Issue by michael-brandt-cu** [conda/conda-build#1204]
We have a package that uses `include_package_data=True` in `setup.py`. With conda-build 1.21.7 and earlier versions, `conda build` works fine.

With 1.21.9 and later versions (1.21.8 is not available via anaconda.org so I couldn't test it), `conda build` was broken, showing this error message in the output:

```
Adding nasateam 0.3.0 to easy-install.pth file

Installed /opt/miniconda/envs/_build_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.5/site-packages/nasateam-0.3.0-py3.5.egg
Processing dependencies for nasateam==0.3.0
Finished processing dependencies for nasateam==0.3.0
Traceback (most recent call last):
  File "/opt/miniconda/envs/_build_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.5/runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "/opt/miniconda/envs/_build_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/miniconda/envs/_build_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.5/py_compile.py", line 186, in <module>
    sys.exit(main())
  File "/opt/miniconda/envs/_build_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.5/py_compile.py", line 178, in main
    compile(filename, doraise=True)
  File "/opt/miniconda/envs/_build_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.5/py_compile.py", line 122, in compile
    source_bytes = loader.get_data(file)
  File "<frozen importlib._bootstrap_external>", line 818, in get_data
FileNotFoundError: [Errno 2] No such file or directory: 'lib/python3.5/site-packages/nasateam-0.3.0-py3.5.egg/nasateam/__init__.py'
```

I have got `conda build` working again by adding `preserve_egg_dir: True` under `build:` in our `recipe/meta.yaml` file.

Despite finding a solution, I wanted to create the issue here because the intended behavior is not 100% clear. Nothing jumps out at me in the release notes for 1.21.8 or 1.21.9 as something that would have caused this. Was this an intended breaking change? Should we have been using `preserve_egg_dir` the whole time?

**Comment by msarahan** [comment]
Thanks for bringing this up.  I'm not sure exactly where the change happened.  Seems like #1155 might have done something like this, but that came later.

Here's the deal, though: eggs imply .pth files, and .pth files are something that we have found to be troublesome.  Conda has a very simple model: whatever is in a package gets extracted.  If files need to be modified, rather than just simply written, then conda breaks.  We use post-link scripts in those situations, and they're brittle, because it's very hard or impossible to generalize what the user's system state is going to be.

So, what we encourage is to avoid eggs and .pth files.  When you install python packages in your conda recipe, and your python package uses setuptools, you should always tell setuptools to not create eggs.  There are two ways I know of:

```
python setup.py install --single-version-externally-managed --record=record.txt
```

```
pip install --no-deps .
```

There is one caveat for pip: on Windows, pip makes entry point exe's that embed the entry point script in the EXE.  conda and conda-build have only very recently begun to support this.  On Linux/Mac, things should be fine with pip.

What you point out is an issue, for sure, and I'd like to fix it one day.  I'll probably need a compelling case where someone needs eggs/.pth files, in order for this issue to have enough priority to get on my docket.  PR's welcome, of course.

**Comment by msarahan** [comment]
I believe this is fixed in 1.21.14.  There was a bug in the ordering of compiling pyc files.  Removing the egg directory was happening before compiling, then compiling went looking for files that were not there.  I'm closing this issue as fixed for now, but please reopen it or file a new issue if your problem remains.
